ivadomed/transforms.py

```diff
@@ -884,21 +908,35 @@ def get_preprocessing_transforms(transforms):
 
 
 def apply_preprocessing_transforms(transforms, seg_pair, roi_pair=None):
+    """
+    Applies preprocessing transforms to segmentation pair (input, gt and metadata).
+    :param transforms: preprocessing transforms (Compose object)
+    :param seg_pair: Segmentation pair containing input and gt
+    :param roi_pair: Segementation pair containing input and roi
+    :return: tuple of seg_pair and roi_pair
+    """
     if transforms is None:
         return (seg_pair, roi_pair)
 
+    metadata_input = seg_pair['input_metadata']
     if roi_pair is not None:
         stack_roi, metadata_roi = transforms(sample=roi_pair["gt"],
                                              metadata=roi_pair['gt_metadata'],
                                              data_type="roi")
+        metadata_input = imed_loader_utils.update_metadata(metadata_roi, metadata_input)
     # Run transforms on images
     stack_input, metadata_input = transforms(sample=seg_pair["input"],
-                                             metadata=seg_pair['input_metadata'],
+                                             metadata=metadata_input,
                                              data_type="im")
     # Run transforms on images
-    stack_gt, metadata_gt = transforms(sample=seg_pair["gt"],
-                                       metadata=seg_pair['gt_metadata'],
-                                       data_type="gt")
+    if seg_pair['gt_metadata'][0] is not None:
+        metadata_gt = imed_loader_utils.update_metadata(metadata_input, seg_pair['gt_metadata'])
+        stack_gt, metadata_gt = transforms(sample=seg_pair["gt"],
+                                           metadata=metadata_gt,
+                                           data_type="gt")
+    else:
+        stack_gt, metadata_gt = seg_pair["gt"], []
```

Comment:
Have you try using this option?
https://github.com/neuropoly/ivadomed/blob/master/ivadomed/transforms.py#L120

Reply:
I am not sure I understand :). Are you asking if it respects the conditions in the line you cited? if so, yes it falls under these conditions since `len(metadata) == 0`

Reply:
As it falls under these conditions: why do we need "if seg_pair['gt_metadata'][0] is not None: // else"?

Reply:
I removed the `if ... else` condition, but I had to add this https://github.com/neuropoly/ivadomed/pull/240/files#diff-05cfa4695be522c97dc494a15bf08959R280 in the clean metadata function. It is slightly cleaner. In a future PR, we might want to clean this. Instead of using None or [None] we could use [] for empty GTs this way, we can remove the `if is not None` condition since the the for loops the empty list will simply be ignored